pkgs/applications/science/misc/root/sw_vers.patch

```diff
@@ -0,0 +1,43 @@
+diff --git a/cmake/modules/SetUpMacOS.cmake b/cmake/modules/SetUpMacOS.cmake
+--- a/cmake/modules/SetUpMacOS.cmake
++++ b/cmake/modules/SetUpMacOS.cmake
+@@ -12,13 +12,9 @@ set(CMAKE_PREFIX_PATH ${CMAKE_PREFIX_PATH} /var/empty/X11R6)
+ #---------------------------------------------------------------------------------------------------------
+ 
+ if (CMAKE_SYSTEM_NAME MATCHES Darwin)
+-  EXECUTE_PROCESS(COMMAND sw_vers "-productVersion"
+-                  COMMAND cut -d . -f 1-2
+-                  OUTPUT_VARIABLE MACOSX_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE)
+-  MESSAGE(STATUS "Found a Mac OS X System ${MACOSX_VERSION}")
+-  EXECUTE_PROCESS(COMMAND sw_vers "-productVersion"
+-                  COMMAND cut -d . -f 2
+-                  OUTPUT_VARIABLE MACOSX_MINOR OUTPUT_STRIP_TRAILING_WHITESPACE)
++  SET(MACOSX_VERSION "10.11")
```

Comment:
The effect of this version is that it enables ```-Wno-deprecated-declarations``` for some dependencies on newer OSX. We don't want to let it use system libraries anyways.

https://github.com/root-mirror/root/blob/0c412b452fb383fad7eda2ffdd9f2e170f72cb24/cmake/modules/SetUpMacOS.cmake#L49-L51
https://github.com/root-mirror/root/blob/0c412b452fb383fad7eda2ffdd9f2e170f72cb24/net/rpdutils/CMakeLists.txt#L19-L21

And other effect is that it sets ```-mmacosx-version-min=${MACOSX_VERSION}```
https://github.com/root-mirror/root/blob/0c412b452fb383fad7eda2ffdd9f2e170f72cb24/cmake/modules/SetUpMacOS.cmake#L138

I guess, I will change the value to ```10.5```.